lib/filewatch/watched_file.rb

```diff
@@ -89,8 +89,27 @@ def file_seek(amount, whence = IO::SEEK_SET)
     end
 
     def file_read(amount)
-      set_accessed_at
-      @file.sysread(amount)
+      #debug "*** file_read #{path}"
+      cc = 3        # max attempts
+      dt = 1.0/128  # delay beetwen attempts
+      loop do
+        set_accessed_at
+        buf = @file.sysread(amount)
+        # return if no zero byte inside
+        return buf unless zc = buf.index("\0")
```

Comment:
The `zc` variable is bound, but unused.

Additionally, for readability we prefer that variables are descriptive and unabiguous, and while I can tell what `buf` is, variables like `cc` and `dt` would be better off with more descriptive names like `attempts_remaining` and `backoff_delay`.